dashboard/app/models/script.rb

```diff
@@ -1631,4 +1633,100 @@ def all_descendant_levels
     sublevels = levels.map(&:all_descendant_levels).flatten
     levels + sublevels
   end
+
+  def serialize_seeding_json
+    eager_loaded_script_levels = ScriptLevel.includes(:levels).where(script_id: id)
+    eager_loaded_levels = eager_loaded_script_levels.map(&:levels).flatten
+
+    scope = {
+      script: self,
+      script_name: name,
+      lesson_groups: lesson_groups,
+      lessons: lessons,
+      script_levels: eager_loaded_script_levels,
+      levels: eager_loaded_levels,
+      levels_script_levels: levels_script_levels
+    }
+
+    {
+      script: ScriptSeed::ScriptSerializer.new(self).as_json,
+      lesson_groups: lesson_groups.map {|lg| ScriptSeed::LessonGroupSerializer.new(lg, scope: scope).as_json},
+      lessons: lessons.map {|l| ScriptSeed::LessonSerializer.new(l, scope: scope).as_json},
+      script_levels: script_levels.map {|sl| ScriptSeed::ScriptLevelSerializer.new(sl, scope: scope).as_json},
+      levels_script_levels: levels_script_levels.map {|lsl| ScriptSeed::LevelsScriptLevelSerializer.new(lsl, scope: scope).as_json}
+    }
+  end
+
+  def self.seed_from_json_file(filename)
```

Comment:
note for later: I think the calls to this will need to be wrapped in a `transaction`, like we do here: https://github.com/code-dot-org/code-dot-org/blob/1e21440030e228afcce54177871367c512e74dd0/dashboard/app/models/script.rb#L888-L889